azure/datalake/store/lib.py

```diff
@@ -269,22 +119,19 @@ class DatalakeRESTInterface:
         'REMOVEDEFAULTACL': ('put', set(), set())
     }
 
-    def __init__(self, store_name=default_store, token=None,
-                 url_suffix=default_adls_suffix, api_version='2018-09-01', req_timeout_s=60, **kwargs):
+    def __init__(self, store_name=default_store, token_credential=None,
```

Comment:
Is token credential the only credential type you support? e.g. do you support api key auth?

Reply:
Seems like api_version & req_timeout_s should be keyword only arguments.

and removing **kwargs is a breaking change.

Is this by design?

Reply:
Breaking changes are fine, as we are making other breaking changes as well(like tokencredential and scope). 

I removed the kwargs as it was mainly forwarding the arguments earlier to the auth function. 

But see your point on making api_version and req_timeout_s keyword only args.